### PR TITLE
Replace deprecated rclcpp::spin_some

### PR DESCRIPTION
**Pull request by christophfroehlich**
This hopefully fixes the CI here, and unblocks #143 from @marqrazz

### CODE REVIEW

File: topic_tools/test/test_demux.cpp [hunk 1]
```diff
@@ -57,7 +58,16 @@ class DemuxTest : public TestTopicToolMultiPub
...
+    const auto timeout = std::chrono::steady_clock::now() + 1s;
+    while (result.wait_for(0s) != std::future_status::ready &&
+      std::chrono::steady_clock::now() < timeout)
+    {
+      executor_->spin_node_all(target_node_, std::chrono::nanoseconds(0));
+      executor_->spin_some();
+    }
+
+    ASSERT_EQ(result.wait_for(0s), std::future_status::ready);
+    ASSERT_TRUE(result.get()->success);
```
Comment (MichaelOrlov) [inline topic_tools/test/test_demux.cpp]: In ROS 2 it is recommended to use `executor_->spin_until_future_complete()` for service requests. It might be corner cases when response from the service server side may be missed if using spin_some. Please refer to my recent fix related to this in the Rosbag2 tests in https://github.com/ros2/rosbag2/pull/2368 and the relevant issue with the Root Cause Analysis in the https://github.com/ros2/rosbag2/issues/2367#issuecomment-4057689265

Comment (christophfroehlich) [reply topic_tools/test/test_demux.cpp]: Thanks for the hint, I applied similar changes to the two files.

**Review by MichaelOrlov — APPROVED** [review]
LGTM with green CI.
